lib/rspec/core/configuration_options.rb

```diff
@@ -195,6 +204,16 @@ def global_options_file
         nil
         # :nocov:
       end
+
+      def global_xdg_options_file
+        xdg_config_home = ENV.fetch("XDG_CONFIG_HOME", "~/.config")
+        File.join(File.expand_path(xdg_config_home), "rspec", "options")
+      rescue ArgumentError
+        # :nocov:
+        RSpec.warning "Unable to find $XDG_CONFIG_HOME/rspec/options because the HOME environment variable is not set"
```

Comment:
What does the `HOME` environment variable have to do with `XDG_CONFIG_HOME`?

Also, I doubt we want to warn if `XDG_CONFIG_HOME` is not set; my impression is that most users won't have it set (and won't care), so printing a warning will just be noisy and confusing for most users.

Reply:
It's not warning when it's not set.
This happens if Ruby fails to `expand_path` the value of the variable, *or* the fallback value of `~/.config`.
The bahavior is identical to the method dealing with `~/.rspec`.

Reply:
Verified that this error would only happen with a weird variable set, but can we improve the message? This warning will only occur when `$XDG_CONFIG_HOME` is malformed.

Reply:
@myronmarston

> What does the HOME environment variable have to do with XDG_CONFIG_HOME?

Not speaking to this particular verbiage, but `HOME/.config` is the default value for `XDG_CONFIG_HOME`.

From the spec:
> If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

Reply:
It's still unclear to me what method call is triggering the `ArgumentError`, which makes it hard to understand in what condition the warning is issued.  Can you refactor so the rescue is directly around the method call that can raise `ArgumentError`?

Reply:
@myronmarston it's the call to `File.expand_path`, Ruby will raise an `ArgumentError` trying to expand `~` if the `HOME` environment variable is not set.

Reply:
According to the tests, this is also something that only happens on Ruby < 2.4.

https://github.com/rspec/rspec-core/blob/17485dea7830b4b8f52f414e043f5743766977a8/spec/rspec/core/configuration_options_spec.rb#L7-L14